### ISSUE TITLE
[istio-operator] istioctl operator init not working

### ISSUE THREAD

**Issue by StefanCenusa** [istio/istio#24815]
**Bug description**
I am following the operator install [instructions](https://istio.io/latest/docs/setup/install/standalone-operator/) on a clean kubernetes 1.17 cluster made with kubespray.

I ran the following commands:
```
➜ istioctl operator init
Using operator Deployment image: docker.io/istio/operator:1.6.3
- Processing resources for Istio operator. Waiting for Deployment/istio-operator/istio-operator
```

The `istio-operator` namespace was created, but the istio-operator pod was in error state
```
NAME                              READY   STATUS                       RESTARTS   AGE
istio-operator-8494bc7758-wr954   0/1     CreateContainerConfigError   0          16m
```
with the following events:

```
Events:
  Type     Reason     Age                  From                     Message
  ----     ------     ----                 ----                     -------
  Normal   Scheduled  <unknown>            default-scheduler        Successfully assigned istio-operator/istio-operator-8494bc7758-wr954 to node-2
  Normal   Pulling    17m                  kubelet, node-2  Pulling image "docker.io/istio/operator:1.6.3"
  Normal   Pulled     16m                  kubelet, node-2  Successfully pulled image "docker.io/istio/operator:1.6.3"
  Warning  Failed     14m (x12 over 16m)   kubelet, node-2  Error: container has runAsNonRoot and image will run as root
  Normal   Pulled     119s (x69 over 16m)  kubelet, node-2  Container image "docker.io/istio/operator:1.6.3" already present on machine
```

Any ideas why the operator container runs as root or, at least, tries to? 

Thank you!

[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

The operator to be installed successfully.

**Steps to reproduce the bug**

Described above.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

```
➜ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-13T18:08:14Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.6", GitCommit:"d32e40e20d167e103faf894261614c5b45c44198", GitTreeState:"clean", BuildDate:"2020-05-20T13:08:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

```
➜ istioctl version --remote=false
1.6.3
```
**How was Istio installed?**

**Environment where bug was observed (cloud vendor, OS, etc)**
On-Premise

**Comment by sdake** [comment]
@StefanCenusa thanks for the bug report. I have attempted to reproduce this issue with bare metal running Kubeadm + metallb + istio 1.6.3. Things work well there.

It is possible you have PSPs in your system that prevent processes from running as root.

I will look into running the operator as a non-root user. I don't think it should require root, but it will require a little bit of testing.

You will face further challenges if your environment does not permit root pods to run. Have you had Istio running in any condition using kubespray?

Cheers,
-steve

**Comment by StefanCenusa** [comment]
@sdake Thanks for the quick answer! 
I am 100% sure that the events errors were caused by PSPs that were installed by default in my cluster and prevented containers to run as root. 
However, I find it odd that an operator requires running as root, thus my initial question. I can totally understand the use-case for the current istio init container to require root, but the other components shouldn't. This kind of policies (running as nonRoot where root it's not absolutely needed) are recommended by most k8s security benchmark tools.

I have had run Istio previously, but on older k8s versions that were not configured with PSPs. I will try to modify my current PSPs to allow istio components to run.

**Comment by sdake** [comment]
@morvencao I don't think https://github.com/istio/istio/pull/24855 is quite right. I had worked on this a few days ago - didn't have a chance to fully test my PR - but thought I'd submit it so the change is done crorectly.

See (untested PR) https://github.com/istio/istio/pull/24858. if you wish to take over this assigned work (I self-assigned a few days ago) - that's fine. Please make sure that the UID/GID are aligned with where the project is headed (distroless) - see issue: https://github.com/istio/istio/issues/24857

Cheers,
-steve

**Comment by sdake** [comment]
@StefanCenusa I agree that no process needs to be run as root - with exception of istio-init. Its a bit of a problem this container runs as root, as the process itself has access in K8s to many different resources.

We will resolve this problem shortly, I am uncertain if a backport will follow.

Cheers,
-steve

**Comment by sdake** [comment]
https://github.com/istio/istio/pull/24960

https://github.com/istio/istio/pull/24929

https://github.com/istio/istio/pull/24963

Thank you for the bug report, this will be available in Istio 1.7.

Cheers,
-steve
